ydb/library/yql/udfs/common/hyperscan/hyperscan_udf.cpp

```diff
@@ -201,7 +201,13 @@ namespace {
             }
 
             if (args[0]) {
-                const std::string_view input(args[0].AsStringRef());
+                // XXX: StringRef data might not be a NTBS, though the function
+                // <TRegExMatch::Match> expects ASCIIZ string. Explicitly copy
+                // the given argument string and append the NUL terminator to it.
+                // See YQL-15189 for more info.
+                const TStringRef inputRef = args[0].AsStringRef();
+                TString input(inputRef.Data(), inputRef.Size() + 1);
```

Comment:
TString input(inputRef.Data(), inputRef.Size()) is enough

Reply:
Fixed, thanks.